### PR TITLE
build: update grpc to 1.34.0

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

Commit Message: Updates gRPC dependency to version 1.33.2. Also updates upb as a requirement, and removes a stale patch.
Additional Description:
-  Side effect, fixes fuzz bug.
- gRPC adds frame size negotiation for ALTS. Used default.

Risk Level: Low, but not non-existent? There was a gRPC default change.
Testing: Added regression test for fuzz bug.
Fixes:
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=27307

### CODE REVIEW

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by asraa** [comment]
Looks like there was a failure in test, after fixing, it seems there's a fake handshake shutdown failure due to a timeout. investigating.

**Comment by asraa** [comment]
@markdroth Hi Mark, quick question. I'm trying to update gRPC and ran into a timeout failure in `IpVersions/AltsIntegrationTestClientInvalidPeer.ClientValidationFail/IPv4` when trying to shutdown the fake handshake server. Just curious, do you have a lead on this? Tried to check if there was anything different about `Shutdown` from 1.27 to 1.33, and couldn't find an immediate lead. Any idea here to start looking? Seems like `Shutdown` never returns. Thank you!

**Comment by moderation** [comment]
I ran into these same timeout issues on a prior attempt at https://github.com/envoyproxy/envoy/pull/13394. The comment https://github.com/envoyproxy/envoy/pull/13394#issuecomment-706363562 from @yashykt about the changed status codes looks like the closest thing I could find. We then decided to defer gRPC and bump protobuf independently

**Comment by asraa** [comment]
> The gRPC commit for the status changes is at https://github.com/grpc/grpc/pull/22901/files

Thanks @moderation! I changed the status code already, which fixed some test issues, but not the `Shutdown()` timeout, which seemed independent. Let me trace through and see if `onRemoteClose()` is being called anywhere there or if I can relate it.

**Comment by asraa** [comment]
@jiangtaoli2016 @yihuazhang Sorry for asking again, I'm attempting to bump gRPC again. The HTTP response code conversions are all fixed, but the timeout remains in the ALTS integration test.

I verified that the client validation actually fails, and that the issue is that `Shutdown()` never returns. I experimented with adding an deadline like `Shutdown(now)`, but no luck. The handshake seems to fail `Couldn't find peer's service account in peer_service_accounts: invalid_client_identity`, but I also `TSI: Handshake successful: peer properties: 5`


```
[2020-11-24 19:02:26.496][28][debug][conn_handler] [source/server/connection_handler_impl.cc:503] [C1] new connection
[2020-11-24 19:02:26.496][28][trace][connection] [source/common/network/connection_impl.cc:517] [C1] socket event: 2
[2020-11-24 19:02:26.496][28][trace][connection] [source/common/network/connection_impl.cc:625] [C1] write ready
[2020-11-24 19:02:26.496][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C1] TSI: doHandshake
[2020-11-24 19:02:26.496][12][trace][connection] [source/common/network/connection_impl.cc:517] [C0] socket event: 2
[2020-11-24 19:02:26.496][12][trace][connection] [source/common/network/connection_impl.cc:625] [C0] write ready
[2020-11-24 19:02:26.496][12][debug][connection] [source/common/network/connection_impl.cc:634] [C0] connected
[2020-11-24 19:02:26.496][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:54] [C0] TSI: doHandshake next: received: 0
[2020-11-24 19:02:26.497][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C0] TSI: doHandshake
[2020-11-24 19:02:26.498][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:79] [C0] TSI: doHandshake next done: status: 0 to_send: 10
[2020-11-24 19:02:26.498][12][trace][connection] [source/common/network/raw_buffer_socket.cc:68] [C0] write returns: 10
[2020-11-24 19:02:26.498][28][trace][connection] [source/common/network/connection_impl.cc:517] [C1] socket event: 3
[2020-11-24 19:02:26.498][28][trace][connection] [source/common/network/connection_impl.cc:625] [C1] write ready
[2020-11-24 19:02:26.498][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C1] TSI: doHandshake
[2020-11-24 19:02:26.498][28][trace][connection] [source/common/network/connection_impl.cc:555] [C1] read ready. dispatch_buffered_data=false
[2020-11-24 19:02:26.498][28][trace][connection] [source/common/network/raw_buffer_socket.cc:25] [C1] read returns: 10
[2020-11-24 19:02:26.498][28][trace][connection] [source/common/network/raw_buffer_socket.cc:39] [C1] read error: Resource temporarily unavailable
[2020-11-24 19:02:26.498][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:169] [C1] TSI: raw read result action 1 bytes 10 end_stream false
[2020-11-24 19:02:26.498][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C1] TSI: doHandshake
[2020-11-24 19:02:26.498][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:54] [C1] TSI: doHandshake next: received: 10
[2020-11-24 19:02:26.499][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:79] [C1] TSI: doHandshake next done: status: 0 to_send: 21
[2020-11-24 19:02:26.499][28][trace][connection] [source/common/network/raw_buffer_socket.cc:68] [C1] write returns: 21
[2020-11-24 19:02:26.499][12][trace][connection] [source/common/network/connection_impl.cc:517] [C0] socket event: 3
[2020-11-24 19:02:26.499][12][trace][connection] [source/common/network/connection_impl.cc:625] [C0] write ready
[2020-11-24 19:02:26.499][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C0] TSI: doHandshake
[2020-11-24 19:02:26.499][12][trace][connection] [source/common/network/connection_impl.cc:555] [C0] read ready. dispatch_buffered_data=false
[2020-11-24 19:02:26.499][12][trace][connection] [source/common/network/raw_buffer_socket.cc:25] [C0] read returns: 21
[2020-11-24 19:02:26.499][12][trace][connection] [source/common/network/raw_buffer_socket.cc:39] [C0] read error: Resource temporarily unavailable
[2020-11-24 19:02:26.499][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:169] [C0] TSI: raw read result action 1 bytes 21 end_stream false
[2020-11-24 19:02:26.499][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C0] TSI: doHandshake
[2020-11-24 19:02:26.499][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:54] [C0] TSI: doHandshake next: received: 21
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:79] [C0] TSI: doHandshake next done: status: 0 to_send: 14
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:100] [C0] TSI: Handshake successful: peer properties: 5
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:103] [C0]   certificate_type: ALTS
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:103] [C0]   service_account: peer_identity
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:103] [C0]   rpc_versions: 

[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:103] [C0]   alts_context: 
peer_identity*local_identity2


[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:103] [C0]   security_level: TSI_PRIVACY_AND_INTEGRITY
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:112] [C0] TSI: Handshake validation failed: Couldn't find peer's service account in peer_service_accounts: invalid_client_identity
[2020-11-24 19:02:26.500][12][debug][connection] [source/common/network/connection_impl.cc:117] [C0] closing data_to_write=0 type=1
[2020-11-24 19:02:26.500][12][debug][connection] [source/common/network/connection_impl.cc:213] [C0] closing socket: 1
[2020-11-24 19:02:26.500][12][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:232] [C0] TSI: closing socket
[2020-11-24 19:02:26.500][12][trace][main] [source/common/event/dispatcher_impl.cc:213] item added to deferred deletion list (size=1)
[2020-11-24 19:02:26.500][28][trace][connection] [source/common/network/connection_impl.cc:517] [C1] socket event: 3
[2020-11-24 19:02:26.500][12][debug][client] [source/common/http/codec_client.cc:96] [C0] disconnect. resetting 0 pending requests
[2020-11-24 19:02:26.500][28][trace][connection] [source/common/network/connection_impl.cc:625] [C1] write ready
[2020-11-24 19:02:26.500][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:45] [C1] TSI: doHandshake
[2020-11-24 19:02:26.500][28][trace][connection] [source/common/network/connection_impl.cc:555] [C1] read ready. dispatch_buffered_data=false
[2020-11-24 19:02:26.500][12][trace][main] [source/common/event/dispatcher_impl.cc:90] clearing deferred deletion list (size=1)
[2020-11-24 19:02:26.500][28][trace][connection] [source/common/network/raw_buffer_socket.cc:25] [C1] read returns: 0
[2020-11-24 19:02:26.500][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:169] [C1] TSI: raw read result action 1 bytes 0 end_stream true
[2020-11-24 19:02:26.500][28][debug][connection] [source/common/network/connection_impl.cc:602] [C1] remote close
[2020-11-24 19:02:26.500][28][debug][connection] [source/common/network/connection_impl.cc:213] [C1] closing socket: 0
[2020-11-24 19:02:26.500][28][debug][connection] [source/extensions/transport_sockets/alts/tsi_socket.cc:232] [C1] TSI: closing socket
[2020-11-24 19:02:26.500][28][trace][main] [source/common/event/dispatcher_impl.cc:213] item added to deferred deletion list (size=1)
[2020-11-24 19:02:26.500][28][debug][conn_handler] [source/server/connection_handler_impl.cc:164] [C1] adding to cleanup list
[2020-11-24 19:02:26.500][28][trace][main] [source/common/event/dispatcher_impl.cc:213] item added to deferred deletion list (size=2)
[2020-11-24 19:02:26.500][28][trace][main] [source/common/event/dispatcher_impl.cc:213] item added to deferred deletion list (size=3)
[2020-11-24 19:02:26.500][28][trace][main] [source/common/event/dispatcher_impl.cc:90] clearing deferred deletion list (size=3)
[2020-11-24 19:02:28.478][24][info][main] [source/server/drain_manager_impl.cc:70] shutting down parent after drain
[2020-11-24 19:02:31.442][24][debug][main] [source/server/server.cc:196] flushing stats
[2020-11-24 19:02:36.446][24][debug][main] [source/server/server.cc:196] flushing stats
[2020-11-24 19:02:41.451][24][debug][main] [source/server/server.cc:196] flushing stats
[2020-11-24 19:02:46.454][24][debug][main] [source/server/server.cc:196] flushing stats
```

**Comment by asraa** [comment]
Confirmed that https://github.com/grpc/grpc/commit/837c4ce589924640bbc395128ba07cbeffdd5b5c#diff-ff0681fdeb3cda181d2ac2d4b80915cd is not 100% the root cause. With a patch that reverts this commit, the tests crash, so the patch did it's part in preventing the crash. As far as the timeout, still no dice.

**Comment by markdroth** [comment]
@jiangtaoli2016 may be able to help with the ALTS problem.

**Comment by jiangtaoli2016** [comment]
Just back from OOO. It looks like the problem is at the shutdown of of the [FakeHandshakerService](https://github.com/grpc/grpc/blob/master/test/core/tsi/alts/fake_handshaker/fake_handshaker_server.cc). @apolcyn last changed it with [this PR](https://github.com/grpc/grpc/commit/528d50ab1564b86e44e1a30041581d2d0b57e63d).

@asraa Could you revert that PR and try and see if it can shutdown correctly?

**Comment by yihuazhang** [comment]
@jiangtaoli2016 I do not think the PR [https://github.com/grpc/grpc/commit/528d50ab1564b86e44e1a30041581d2d0b57e63d] is the culprit because it is introduced in gRPC version 1.26, and ALTS integration test still works with version 1.27.

**Comment by asraa** [comment]
update: @jiangtaoli2016 helped debug and fixed the shutdown happening in client validation fail. change pushed

however, another timeout is happening in `CheckAltsVersion`. https://github.com/envoyproxy/envoy/blob/5e2a0e6a5c03774bf4d12ac2ebddfa59631a1a27/test/extensions/transport_sockets/alts/alts_integration_test.cc#L60 is stuck in an infinite loop waiting for a client request. will get back to this later.

**Comment by yihuazhang** [comment]
@asraa, regarding the timeout happened in `CheckAltsVersion`, I think after extracting the RPC version from the first client and server handshake message at line (https://github.com/envoyproxy/envoy/blob/master/test/extensions/transport_sockets/alts/alts_integration_test.cc#L73), it should return OK, instead of hang there.

**Comment by asraa** [comment]
@yihuazhang Thank you so much! Tests pass :)

**Comment by htuch** [comment]
@asraa can you take a look at format etc?

**Comment by moderation** [comment]
@asraa gRPC 1.34.0 is out if you want to try and jump all the way to latest - https://github.com/grpc/grpc/releases/tag/v1.34.0

**Comment by asraa** [comment]
Will do! I'm not sure why I keep hitting CI errors. Looks like one after the merge.

**Comment by asraa** [comment]
Bumped to 1.34.0. It looks like I'm hitting this https://github.com/envoyproxy/envoy/blob/f8b234bb3c416d04ae940ff3cab8649dade78841/source/common/grpc/async_client_impl.cc#L113 in clang_tidy.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -223,14 +223,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        version = "d8f4928fa779f6005a7fe55a176bdb373b0f910f",
-        sha256 = "bbc8f020f4e85ec029b047fab939b8c81f3d67254b5c724e1003a2bc49ddd123",
+        version = "1.34.0",
+        sha256 = "7372a881122cd85a7224435a1d58bc5e11c88d4fb98a64b83f36f3d1c2f16d39",
         strip_prefix = "grpc-{version}",
-        urls = ["https://github.com/grpc/grpc/archive/{version}.tar.gz"],
+        urls = ["https://github.com/grpc/grpc/archive/v{version}.tar.gz"],
         use_category = ["dataplane_core", "controlplane"],
-        release_date = "2020-02-11",
+        release_date = "2020-12-01",
```
Comment (moderation) [inline bazel/repository_locations.bzl:231]: The UTC `release_date` for this release is `2020-12-02` but we haven't landed the updated date checker which checks the release date vs. the date of the last commit in the release. This is FYI and I can clean up once the new script lands.

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -774,12 +772,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         project_url = "https://github.com/protocolbuffers/upb",
-        version = "8a3ae1ef3e3e3f26b45dec735c5776737fc7247f",
-        sha256 = "e9f281c56ab1eb1f97a80ca8a83bb7ef73d230eabb8591f83876f4e7b85d9b47",
+        version = "382d5afc60e05470c23e8de19b19fc5ad231e732",
+        sha256 = "7992217989f3156f8109931c1fc6db3434b7414957cb82371552377beaeb9d6c",
         strip_prefix = "upb-{version}",
         urls = ["https://github.com/protocolbuffers/upb/archive/{version}.tar.gz"],
         use_category = ["controlplane"],
-        release_date = "2019-11-19",
+        release_date = "2020-08-03",
```
Comment (moderation) [inline bazel/repository_locations.bzl:780]: Latest `upb` if you want to use that is:
```
    upb = dict(
        project_name = "upb",
        project_desc = "A small protobuf implementation in C (gRPC dependency)",
        project_url = "https://github.com/protocolbuffers/upb",
        version = "794ce6d06188c6df290b4b46b4c32e95a219eafc",
        sha256 = "1920bdcb3b963db424ce0a0e6cdcc9d5b5d48ef3c95bb79823ba3c8f63422d01",
        release_date = "2020-11-26",
        strip_prefix = "upb-{version}",
        urls = ["https://github.com/protocolbuffers/upb/archive/{version}.tar.gz"],
        use_category = ["controlplane"],
        cpe = "N/A",
    ),
```

**Comment by asraa** [comment]
CI is passing besides the existing clang_tidy issue

**Comment by asraa** [comment]
Upstream merged, clang-tidy is still pre-existing.

**Comment by asraa** [comment]
@mattklein123 @moderation Sorry for bumping this again, but merged in main, clang-tidy is still failing on existing TODO

**Comment by mattklein123** [comment]
OK let's force merge given it's an existing issue.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**